src/mixins/OFTWrapper.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.8.0;
+
+import "openzeppelin-contracts/security/ReentrancyGuard.sol";
+import "openzeppelin-contracts/access/Ownable.sol";
+import "openzeppelin-contracts/token/ERC20/utils/SafeERC20.sol";
+import "layerzerolabs-solidity-examples/token/oft/v2/IOFTV2.sol";
+import "layerzerolabs-solidity-examples/token/oft/IOFT.sol";
+import "interfaces/IOFTWrapper.sol";
+import "interfaces/IAggregator.sol";
+
+contract OFTWrapper is IOFTWrapper, Ownable, ReentrancyGuard {
+    using SafeERC20 for IOFT;
+
+    IAggregator public aggregator;
+    IOFTV2 public immutable oft;
+    IOFT public immutable token;
+
+    uint256 public defaultExchangeRate;
+
+    error InvalidQuoteType();
+
+    constructor(uint256 _defaultExchangeRate, address _oft, address _aggregator, address _multisig) {
+        defaultExchangeRate = _defaultExchangeRate;
+        require(_oft != address(0), "OFTWrapper: invalid oft");
+        oft = IOFTV2(_oft);
+        token = IOFT(oft.token());
+        require(_aggregator != address(0), "OFTWrapper: invalid aggregator");
+        aggregator = IAggregator(_aggregator);
+        transferOwnership(_multisig);
+    }
+
+    function setDefaultExchangeRate(uint256 _defaultExchangeRate) external onlyOwner {
```

Comment:
missing event here
ex: emit LogDefaultExchangeRateChanged(old, new)

I usually emit before changing the storage variable to avoid creating a temp variable